experiments/calibration/shared/observation_data_locations.jl

```diff
@@ -0,0 +1,41 @@
+era5_ds = Dataset(
+    joinpath(
+        ClimaLand.Artifacts.era5_surface_data2008_path(),
+        "era5_monthly_surface_fluxes_200801-200812.nc",
+    ),
+)
+
+# Make the ERA5 target
+ERA5_target = []
+close = (x, y) -> abs(x - y) < 5e-1
+for (lon, lat) in training_locations
+    # Fetch slices of lhf and shf era5 data from the era5 dataset
+    lat_ind, lon_ind = findall((x) -> close(x, lat), era5_ds["latitude"][:])[1],
+    findall((x) -> close(x, lon + 180), era5_ds["longitude"][:])[1]
+    lhf_loc = vec(era5_ds["mslhf"][lon_ind, lat_ind, :][:, end, :])
+    shf_loc = vec(era5_ds["msshf"][lon_ind, lat_ind, :][:, end, :])
```

Comment:
Just a check on the relative grid resolutions of the data and the forward model. Do we have both ERA5 and CL interpolated (via local averaging for example) to the same lat-lon grid here?

otherwise if the CL grid is coarser than the ERA5 grid, the value from direct slicing here at a specfic lat-lon in ERA5  will represent a different quantity than CL run at that lat-lon